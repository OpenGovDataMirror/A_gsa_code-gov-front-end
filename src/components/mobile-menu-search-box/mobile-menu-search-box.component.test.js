import React from 'react'
import { shallow } from 'enzyme'

import MobileMenuSearchBox from 'components/mobile-menu-search-box/mobile-menu-search-box.component'

const props = {
  onSubmit: jest.fn(),
  toggleMobileMenu: jest.fn(),
  mobileMenu: true
}

let wrapper
let instance
describe('components - MobileMenuSearchBox', () => {
  beforeEach(() => {
    wrapper = shallow(<MobileMenuSearchBox {...props} />)
    instance = wrapper.instance()
  })

  describe('handleChange', () => {
    beforeEach(() => {
      instance.handleChange({ target: { value: 'test-value' } })
    })

    it('should set the value', () => {
      expect(wrapper.state('value')).toBe('test-value')
    })
  })

  describe('handleSubmit', () => {
    beforeEach(() => {
      wrapper.setState({ value: 'test-value' })
      instance.handleSubmit({ preventDefault: jest.fn() })
    })

    it('should call `onSubmit` with the value', () => {
      expect(props.onSubmit).toBeCalledWith('test-value')
    })

    it('should clear the value', () => {
      expect(wrapper.state('value')).toBe('')
    })

    it('should call `toggleMobileMenu` with props.mobileMenu', () => {
      expect(props.toggleMobileMenu).toBeCalledWith(props.mobileMenu)
    })
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
