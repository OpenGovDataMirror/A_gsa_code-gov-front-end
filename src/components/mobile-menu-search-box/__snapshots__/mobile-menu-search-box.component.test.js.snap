// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - MobileMenuSearchBox render should render correctly 1`] = `
<Fragment>
  <form
    className="usa-search"
    onSubmit={[Function]}
    role="search"
  >
    <input
      aria-label="search"
      className="usa-input radius-left-md border-1px border-right-0 border-base-lighter font-body-3xs"
      onChange={[Function]}
      placeholder="Search Projects..."
      type="search"
      value=""
    />
    <button
      className="usa-button font-body-sm"
      type="submit"
    >
      <span
        className="usa-search__submit-text"
      >
        Go
      </span>
    </button>
  </form>
</Fragment>
`;
